doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -603,15 +607,37 @@ def generate_payload_entries(self, assembly_inspector: AssemblyInspector) -> (Di
                 continue
             # No adjustment for private or public; the assembly's canonical payload content is the same.
 
-            entries: Dict[str, PayloadEntry]  # Key of this dict is release payload tag name
-            issues: List[AssemblyIssue]
-            entries, issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PUBLIC))
+            # public_entries: Dict[str, PayloadEntry]  # Key of this dict is release payload tag name
+            # public_payload_issues: List[AssemblyIssue]
+            entries, payload_issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PUBLIC))
+
+            public_entries: Dict[str, PayloadEntry] = dict()
+            for k, v in entries.items():
+                if not v.build_inspector:
+                    # Its RHCOS, since it doesn't have a build inspector. Put it in for now, but change once RHCOS
+                    # supports private nightlies
+                    public_entries[k] = v
+                    continue
+
+                if v.build_inspector.is_under_embargo() and self.runtime.assembly_type == AssemblyTypes.STREAM:
+                    # It's an embargoed build. Filter it out if its stream
+                    continue
+
+                public_entries[k] = v
 
-            # Exclude issues for now, for private repo
-            private_entries, _ = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PRIVATE))
             entries_for_arch[arch] = entries
+            self.assembly_issues.extend(payload_issues)
+
+            # Report issues for any embargoed content being made public.
+            # If releasing after embargo lift, these can be permitted using 'EMBARGOED_CONTENT' code
+            embargo_issues = PayloadGenerator.embargo_issues_for_payload(public_entries, arch)
+            self.assembly_issues.extend(embargo_issues)
+
+            private_entries, private_payload_issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PRIVATE))
             private_entries_for_arch[arch] = private_entries
-            self.assembly_issues.extend(issues)
+            # We do not want to add private_payload_issues into the assembly issues list because it will just be the same issues as public, so we'll get duplicates
+            # This can be uncommented if it changes in the future
+            # self.assembly_issues.extend(private_payload_issues)
```

Comment:
I think this will miss If there is an issue with a private build, so this should get us only the private build issues
```suggestion
            private_only_issues = set(private_payload_issues) - set(payload_issues)
            self.assembly_issues.extend(list(private_only_issues))
```